src/components/Api.js

```diff
@@ -0,0 +1,129 @@
+class Api {
+  constructor(options) {
+    this.options = options;
+  }
+
+  // Common method for handling fetch requests
+  _handleResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return res.json().then((data) => Promise.reject(data));
+  }
+
+  // Common method for handling fetch errors
+  _handleError(err) {
+    console.error(`Error: ${err.message}\nURL: ${err.url}`);
+  }
+
+  // Get user information
+  getUserInfo() {
+    return fetch(`${this.options.baseUrl}/users/me`, {
+      headers: {
+        authorization: this.options.headers.authorization,
+      },
+    })
+      .then(this._handleResponse)
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._handleResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.